onnxruntime/core/providers/cpu/math/element_wise_ops.h

```diff
@@ -6,132 +6,246 @@
 #include "core/common/common.h"
 #include "core/framework/op_kernel.h"
 #include "core/util/math_cpuonly.h"
+#include "core/providers/cpu/element_wise_ranged_transform.h"
 
 namespace onnxruntime {
+namespace functors {
 
-template <typename T>
-class Add final : public OpKernel {
- public:
-  Add(const OpKernelInfo& info) : OpKernel(info) {
+template<typename T>
+struct Log : public ElementWiseRangedTransform<T> {
```

Comment:
Nit, maybe to tidy in a separate PR : could order these transforms alphabetically in the file

Reply:
Agree. Will do it in another PR.